avm/src/vm/opcode.rs

```diff
@@ -3658,8 +3672,21 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       let ns = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[0])?)?;
       let key = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[1])?)?;
       let nskey = format!("{}:{}", ns, key);
-      let ds = Arc::clone(&DS);
-      ds.insert(nskey, hm);
+      let ctrl_port = CONTROL_PORT_CHANNEL.get();
+      let ctrl_port = match ctrl_port {
+        Some(ctrl_port) => Some(ctrl_port.borrow().clone()), // TODO: Use thread-local storage
```

Comment:
i can see this cloning being very expensive... is it necessary or can we just `borrow()` it? If we can get rid of it, this whole reassignment can be:
```rust
let ctrl_port = ctrl_port.map(watch::Receiver::borrow);
```

Reply:
I couldn't get it to compile with just `borrow`ing it. But as the `TODO` mentions I want to follow this up with a thread-local storage for the control port that is only updated when there's actually a change to it, so that cost all goes completely away *and* there's no lock contention on the borrow (the `clone` may be expensive, but once it's done, the lock is freed, so other datastore opcodes can also run.